packages/template-typescript-minimal/package.json

```diff
@@ -58,7 +59,8 @@
                     "host": "api.cquotient.com",
                     "path": "einstein"
                 }
-            ]
+            ],
+            "enablePrepass": true
```

Comment:
I'm -1 on this setting – I don't think it'll make sense to anybody and I don't think the fundamentals of our rendering function should be configurable.

I'm guessing you're doing this so that people that aren't using `react-query` don't pay the additional pre-rendering cost? A few things:

1. If you release this, I would expect all existing `getProps` users to want to switch to hooks, since they've been asking for it. So likely everyone would want to opt-in.
2. I think we would want to _recommend_ the hooks instead of `getProps`, right? That being the case, the hooks need to be turned on by default.
3. To make decisions on the API it would be good to know what the performance penalty is. Can we measure with/without `prepass` before we commit?

Reply:
Thats a good point. You're right I was thinking about the cost, but if there isn't one, then yes this options is a complexity we don't need. I'll do some testing.

Reply:
So I did what would be closer to real world testing by integrating with the `template-retail-react-app` as there isn't much happening in the `template-typescript-minimal` application. The result is **~400-500ms** penalty for using `ssrPrepass`. 

My thoughts are that this is significant enough to warrant feature flagging that logic. We don't want customers upgrading their SDK's and then having a 400ms slow down. This feature will eventually become the default once we start integrating it with the retail react app, and move away from the getprops usage.

Reply:
It was noted that during my speed testing of `react-prepass`, the _prepass_ function was running after the `getProps` statics where being invoked. I thought this meant that prepass was traversing components that it didn't necessarily have to traverse. This assumption was wrong, from the looks of it, prepass is smart enough to process unique components.

Either way, after disabling getProps and analyzing the performance, it's apparent that prepass is close to 1-1 in how long it takes to run compared to the actual render itself. You can see from the attach screen capture:
![Screen Shot 2022-08-04 at 2 56 00 PM](https://user-images.githubusercontent.com/8902581/182961377-1ba445b2-feb9-4b92-9c3a-79b90f706e25.png)


What does this mean? It means that to support useQuery we need react prepass, and using react prepass means you effectively double your rendering time. Is it worth it?